apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/AbstractMethodInterceptor.java

```diff
@@ -111,7 +84,7 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
             if (stackDepth == null) {
                 final ContextCarrier contextCarrier = new ContextCarrier();
 
-                if (IN_SERVLET_CONTAINER && IS_JAVAX && HttpServletRequest.class.isAssignableFrom(request.getClass())) {
+                if (isAssignableFrom(JAVAX_SERVLET_REQUEST_CLASS, request.getClass())) {
```

Comment:
If isAssignableFrom executes every time, you will slow the whole process.

Reply:
`static` block is an important mechanism. You should not remove them all.

Reply:
Meanwhile, I think from performance perspective, use the witness class directly should be quicker too.

Reply:
> If isAssignableFrom executes every time, you will slow the whole process.

I've considered the need for a cache before, but then I decided that a class has a very limited number of superclasses, and this is a completely purely in-memory operation that should be very fast

Reply:
No, it doesn't. You could run benchmark, metadata access is not performance oriented.

Reply:
> `static` block is an important mechanism. You should not remove them all.

The top part of the static block looks a bit redundant. Because when it's used, for example:
```java
if (IS_SERVLET_GET_STATUS_METHOD_EXIST && HttpServletResponse.class.isAssignableFrom(response.getClass()))
```
If response's class is a subclass of`HttpServletResponse`, it certainly have `getStatus()` method.

Reply:
> Meanwhile, I think from performance perspective, use the witness class directly should be quicker too.

Using the witness class mechanism to recognize the SDK version is better and should be clearer to read. Since it's an abstract class used by multiple versions of plugins, we would need to do some refactoring to it. Is this consistent with your thoughts?

Reply:
I think for any plugin using jaxax and/or jakata, they need to deal with this kind of conflict, right?
It is better we have a unified way to do that.

Reply:
I believe there should be no conflicts if each plugin defines an accurate witness class. Perhaps removing this cumbersome abstraction and pushing the logic down into each version of the plugins would look more natural. ^^